### PR TITLE
Add button to fit image into current view

### PR DESCRIPTION
**Pull request by precla**
Adds a button so the user can fit the image to the current view.
Does not automatically fit image. Do we want that?

fitimage

as requested in: https://github.com/ksnip/kImageAnnotator/issues/21

### CODE REVIEW

**Comment by precla** [comment]
The approach with `-1` as the `value` is probably not the most straight forward and sane one. The goal was to add/change as few lines of code as possible...
Thoughts?

File: src/gui/scrollAndZoomView/ViewZoomer.cpp [hunk 1]
```diff
@@ -64,8 +64,19 @@ void ViewZoomer::wheelZoom(QWheelEvent *event)
...
 
+void ViewZoomer::fitImage()
+{
+	mView->fitInView(mView->sceneRect(), Qt::KeepAspectRatio);
+}
+
 void ViewZoomer::setZoomValue(double value)
 {
+	// ZoomPicker's mFitImageButton() sets value to -1
+	if (value < 0) {
```
Comment (DamirPorobic) [inline src/gui/scrollAndZoomView/ViewZoomer.cpp:75]: You're right, this is a bit strange. When you think about it, the zoom value has not yet change but rather you want to send a notification from the zoom picker to the ViewZoomer and the ViewZoomer then does something that changes the zoom value and this new zoom value should be sent back to the zoom picker. So maybe a new signal for the zoom picker that the ViewZoomer listens to?

**Review by DamirPorobic — COMMENTED** [review]
Looking good and seems to work but the -1 as message I would replace with a dedicated signal. 

Also something that I've noticed, when I fit the image to view, I get something like 116% zoom value and our zoom increments are 10% so I cannot get back to 100% without using the keyboard shortcut `CTRL+0` which I had forgotten about myself, event though I'm the one who implemented it. What do you think about another button with corresponding signal that resets the view to 100% or do you think that this is overkill? 

Any tests?

**Comment by precla** [comment]
> Looking good and seems to work but the -1 as message I would replace with a dedicated signal.

Ok, makes more sense.

> Also something that I've noticed, when I fit the image to view, I get something like 116% zoom value and our zoom increments are 10% so I cannot get back to 100% without using the keyboard shortcut `CTRL+0` which I had forgotten about myself, event though I'm the one who implemented it. 
> 
> Any tests?

yeah, that sounds reasonable. Fixed so it reverts to 10% in/decrements on manual zoom change:
https://github.com/ksnip/kImageAnnotator/pull/281/files#diff-2130e12e9f11ef241e05aa6fb1aef112d7d530d5780bcd6bdcc9c2d79f5a681cR42

(another button just to get 10% steps would be too much imho)

Tested standalone and in Spectacle. Looks like it always misses by 1% when fitting in Spectacle..

>What do you think about another button with corresponding signal that resets the view to 100% or do you think that this is overkill?

Scrolling back to 100% is fast and easy. Wouldn't be a problem to add (and invoke my icon drawing skills again haha).

File: src/gui/scrollAndZoomView/ViewZoomer.cpp (hunk 1 above)
Comment (precla) [reply src/gui/scrollAndZoomView/ViewZoomer.cpp:76]: That means going trough `AnnotationGeneralSettings`, over `AnnotationSettingsAdapter` and land in `AbstractSettingsProvider` to access `mZoomValueProvider` :D

*hold my beer*

Comment (precla) [reply src/gui/scrollAndZoomView/ViewZoomer.cpp]: Done, added as additional commit instead of `--amend`ing.

**Comment by precla** [comment]
Hm, build failing on github, working fine on my end. :s

**Comment by DamirPorobic** [comment]
Looks like you have changed the ZoomProvider interface but not the Mock that implements it for the tests. I guess you're not building the tests local that's why it builds locally but not on the CI. You need to enable the test build, then you'll see it `-DBUILD_TESTS=ON`

**Comment by precla** [comment]
done with Mock and added 100% , aka `reset zoom` button: 

Screenshot_20220207_222514

**Comment by DamirPorobic** [comment]
Sorry for the delay, merged. I think I'll have a look into the reset icon, it's a bit redundant to have 100% twice there, but for now it's fine. Thanks for providing this PR!

**Comment by precla** [comment]
No problem. What other icon do you mean with reset? Can't see any other

**Comment by DamirPorobic** [comment]
Not sure, maybe something like this?
image

Or maybe having a single button, like extended option and there something like a popup menu opens with "Fit to View" and "Reset". Just thinking aloud.

**Comment by precla** [comment]
Hm, the rotating arrows remind me too much of the 'rotate image' button.
I'll think about it too, and respond if I manage to come up with some better.

**Pull request merged**